### ISSUE TITLE
`InternalServerError: error initiating restore protocol` upgrading a project from 5.0-beta.1 to 5.0-beta.2

### ISSUE THREAD

**Issue by raddevon** [edgedb/edgedb#7150]
I'm having difficulty seeing if this is reproducible with a fresh 5.0-beta.1 database because of https://github.com/edgedb/edgedb-cli/issues/1280, so I haven't provided any instructions to reproduce.


<details>
  <summary>Output</summary>
  
```
Upgrading to a major version 5.0-beta.2+796ed0e
Dumping the database...
Starting dump for edgedb...
  Finished dump for edgedb. Total size: 648 B                                                                                                                                                                                      Restoring the database...

Restoring database from file `/Users/raddevon/Library/Application Support/edgedb/data/edgedb_v5.dump/edgedb.dump`. Total size: 0.04 MB
edgedb error: cannot restore "edgedb_v5": restoring database "edgedb": InternalServerError: error initiating restore protocol
  Hint: This is most likely a bug in EdgeDB. Please consider opening an issue ticket at https://github.com/edgedb/edgedb/issues/new?template=bug_report.md
To undo run:
  edgedb instance revert -I "edgedb_v5"
```

</details>


Running `edgedb instance revert` as suggested produces this:
<details>
  <summary>Output</summary>

```
EdgeDB version: 5.0-beta.1+8477d12
Backup timestamp: 2024-04-03T12:39:47.186403000Z (done 15m ago)
Note: backup appears to be from a broken upgrade

Currently stored data will be lost and overwritten by the backup.
Do you really want to revert? (type `Yes`)
> Yes
Starting EdgeDB 5.0-beta.1+8477d12 ...
Instance edgedb_v5 is successfully reverted to 5.0-beta.1+8477d12
```

</details>

Retrying the upgrade after the revert produces this:
<details>
  <summary>Output</summary>

```
Using 5.0-beta.2 (matches `5.0-beta.1`), use `=5.0-beta.1` for exact version
Version 5.0-beta.2+796ed0e is already downloaded
⠂ checking migrations                                                                                                                                                                                                              Legacy migration file names detected, consider running 'edgedb migration upgrade-format'
The schema is forward compatible. Ready for upgrade.
[edgedb] ---- Exception occurred: SIGTERM ----
[edgedb] 2. asyncio.exceptions.CancelledError:
[edgedb]
[edgedb]
[edgedb] ---- Traceback ----
[edgedb]     /Users/raddevon/Library/Application Support/edgedb/portable/5.0-beta.2/lib/python3.12/site-packages/edb/common/signalctl.py, line 205, in wait_for
[edgedb]         > return fut.result()
[edgedb]
[edgedb]     /Users/raddevon/Library/Application Support/edgedb/portable/5.0-beta.2/lib/python3.12/site-packages/edb/server/server.py, line 1660, in stop
[edgedb]         > await super().stop()
[edgedb]     /Users/raddevon/Library/Application Support/edgedb/portable/5.0-beta.2/lib/python3.12/site-packages/edb/server/server.py, line 1027, in stop
[edgedb]         > await self._stop_servers(self._servers.values())
[edgedb]     /Users/raddevon/Library/Application Support/edgedb/portable/5.0-beta.2/lib/python3.12/site-packages/edb/server/server.py, line 927, in _stop_servers
[edgedb]         > async with asyncio.TaskGroup() as g:
[edgedb]     /Users/raddevon/Library/Application Support/edgedb/portable/5.0-beta.2/lib/python3.12/asyncio/taskgroups.py, line 134, in __aexit__
[edgedb]     /Users/raddevon/Library/Application Support/edgedb/portable/5.0-beta.2/lib/python3.12/asyncio/taskgroups.py, line 110, in __aexit__
[edgedb]         > raise propagate_cancellation_error
[edgedb]         > await self._on_completed_fut
[edgedb]
[edgedb]
[edgedb] asyncio.exceptions.CancelledError:
[edgedb]
[edgedb] ---- Traceback ----
[edgedb] 1. edb.common.signalctl.SignalError: SIGTERM
[edgedb]
[edgedb] ---- The above exception was the direct cause of the following exception ----
[edgedb]
[edgedb]     /Users/raddevon/Library/Application Support/edgedb/portable/5.0-beta.2/bin/edgedb-server.py, line 7, in <module>
[edgedb]         > sys.exit(main())
[edgedb]     /Users/raddevon/Library/Application Support/edgedb/portable/5.0-beta.2/lib/python3.12/site-packages/click/core.py, line 1157, in __call__
[edgedb]     /Users/raddevon/Library/Application Support/edgedb/portable/5.0-beta.2/lib/python3.12/site-packages/click/core.py, line 1078, in main
[edgedb]         > return self.main(*args, **kwargs)
[edgedb]         > rv = self.invoke(ctx)
[edgedb]     /Users/raddevon/Library/Application Support/edgedb/portable/5.0-beta.2/lib/python3.12/site-packages/click/core.py, line 1666, in invoke
[edgedb]         > rv = super().invoke(ctx)
[edgedb]     /Users/raddevon/Library/Application Support/edgedb/portable/5.0-beta.2/lib/python3.12/site-packages/click/core.py, line 1434, in invoke
[edgedb]         > return ctx.invoke(self.callback, **ctx.params)
[edgedb]     /Users/raddevon/Library/Application Support/edgedb/portable/5.0-beta.2/lib/python3.12/site-packages/click/core.py, line 783, in invoke
[edgedb]         > return __callback(*args, **kwargs)
[edgedb]     /Users/raddevon/Library/Application Support/edgedb/portable/5.0-beta.2/lib/python3.12/site-packages/click/decorators.py, line 33, in new_func
[edgedb]         > return f(get_current_context(), *args, **kwargs)
[edgedb]     /Users/raddevon/Library/Application Support/edgedb/portable/5.0-beta.2/lib/python3.12/site-packages/edb/server/main.py, line 777, in main
[edgedb]         > server_main(**kwargs)
[edgedb]     /Users/raddevon/Library/Application Support/edgedb/portable/5.0-beta.2/lib/python3.12/site-packages/edb/server/main.py, line 755, in server_main
[edgedb]         > asyncio.run(run_server(
[edgedb]     /Users/raddevon/Library/Application Support/edgedb/portable/5.0-beta.2/lib/python3.12/asyncio/runners.py, line 194, in run
[edgedb]         > return runner.run(main)
[edgedb]     /Users/raddevon/Library/Application Support/edgedb/portable/5.0-beta.2/lib/python3.12/asyncio/runners.py, line 118, in run
[edgedb]         > return self._loop.run_until_complete(task)
[edgedb]     uvloop/loop.pyx, line 1517, in uvloop.loop.Loop.run_until_complete
[edgedb]
[edgedb]     /Users/raddevon/Library/Application Support/edgedb/portable/5.0-beta.2/lib/python3.12/site-packages/edb/server/main.py, line 655, in run_server
[edgedb]         > await _run_server(
[edgedb]     /Users/raddevon/Library/Application Support/edgedb/portable/5.0-beta.2/lib/python3.12/site-packages/edb/server/main.py, line 336, in _run_server
[edgedb]         > await sc.wait_for(ss.stop())
[edgedb]     /Users/raddevon/Library/Application Support/edgedb/portable/5.0-beta.2/lib/python3.12/site-packages/edb/common/signalctl.py, line 238, in wait_for
[edgedb]         > raise cancelled_by
[edgedb]
[edgedb] edb.common.signalctl.SignalError: SIGTERM
Upgrading to a major version 5.0-beta.2+796ed0e
Dumping the database...
Starting dump for edgedb...
  Finished dump for edgedb. Total size: 648 B
edgedb error: Upgrade is already in progress
```

</details>

Subsequent attempts produce the same.

- EdgeDB Version:
- EdgeDB CLI Version: 4.3.0-dev.1123+13f7e12
- OS Version: macOS 14.0


<!-- If the issue is about a query error, please also provide your schema -->

Schema:
```esdl
module default {
    type Person {
        property name: std::str;
    };
};
```

**Comment by TreeOfLearning** [comment]
I'm also seeing this - although I lost power during my first attempt to upgrade, rather than it getting to the end but failing like in Devon's example. 

Interestingly, when I try to upgrade now, mine doesn't even manage to dump the database before telling me there's an upgrade in progress.

Discord thread [here](https://discord.com/channels/841451783728529451/849377751370432573/1225122861891719330)

**Comment by msullivan** [comment]
Obviously this seems pretty bad but I don't know if this can be reproduced?

**Comment by raddevon** [comment]
Just tried again to reproduce with a newer version of the CLI, and I get this when I try to spin up an instance on 5.0-beta.1:
```
No `edgedb.toml` found in `/Users/raddevon/Documents/projects/edgedb/troubleshooting/5-beta-upgrade` or above
Initializing new project...
Checking EdgeDB versions...
┌─────────────────────┬──────────────────────────────────────────────────────────────────────────────────────┐
│ Project directory   │ /Users/raddevon/Documents/projects/edgedb/troubleshooting/5-beta-upgrade             │
│ Project config      │ /Users/raddevon/Documents/projects/edgedb/troubleshooting/5-beta-upgrade/edgedb.toml │
│ Schema dir (empty)  │ /Users/raddevon/Documents/projects/edgedb/troubleshooting/5-beta-upgrade/dbschema    │
│ Installation method │ portable package                                                                     │
│ Version             │ 5.0-beta.1+8477d12                                                                   │
│ Instance name       │ 5_beta_upgrade                                                                       │
└─────────────────────┴──────────────────────────────────────────────────────────────────────────────────────┘
Version 5.0-beta.1+8477d12 is already downloaded
Initializing EdgeDB instance...
Applying migrations...
[2024-05-01T13:23:16Z WARN  edgedb_tokio::raw::connection] Error received from server: database 'main' does not exist. Severity: Error. Code: 0x4030005
edgedb error: ClientConnectionFailedError: end of stream while reading message
```

I guess we can go ahead and close this since it doesn't seem like it's going to be reproducible under the stable CLI. Not sure it would be worth it to fix whatever is preventing a 5.0-beta.1 instance from being started right now since it's unlikely anyone will want to do that anymore.
